records/generator.go

```diff
@@ -27,9 +29,17 @@ type rrs map[string][]string
 // RecordGenerator contains DNS records and methods to access and manipulate
 // them. TODO(kozyraki): Refactor when discovery id is available.
 type RecordGenerator struct {
-	As       rrs
-	SRVs     rrs
-	SlaveIPs map[string]string
+	As         rrs
+	SRVs       rrs
+	SlaveIPs   map[string]string
+	httpClient http.Client
+}
+
+// NewRecordGenerator returns a RecordGenerator that's been configured with a timeout.
+func NewRecordGenerator(httpTimeout time.Duration) *RecordGenerator {
```

Comment:
I'm still investigating cases where we might want to blow away the httpclient entirely (stale connections, etc..). Originally, I had it where if HTTP timeout was encountered, it'd blow away the old client. I'm not sure how legitimate this concern is, hence I didn't proceed.